### ISSUE TITLE
Getting a message, and also useless-suppression of message.

### ISSUE THREAD

**Issue by nedbat** [PyCQA/pylint#5215]
### Bug description

```python
"""This is test_report.py"""

class FakeReporter:
    """A fake implementation of a one-file reporter."""

    def report(self, morfs):   # pylint: disable=attribute-defined-outside-init
        """Fake."""
        self.morfs = morfs
```

### Configuration

```ini
# pylintrc

[MESSAGES CONTROL]
enable =
    useless-suppression
```

### Command used

```shell
python -m pylint test_report.py
```

### Pylint output

```shell
************* Module test_report
test_report.py:8:8: W0201: Attribute 'morfs' defined outside __init__ (attribute-defined-outside-init)
test_report.py:3:0: R0903: Too few public methods (1/2) (too-few-public-methods)
test_report.py:6:0: I0021: Useless suppression of 'attribute-defined-outside-init' (useless-suppression)

-----------------------------------
Your code has been rated at 3.33/10
```

### Expected behavior

I should be able to suppress the message, and not be told that it is useless to suppress it.

### Pylint version

```shell
$ python -m pylint --version
pylint 2.11.1
astroid 2.8.3
Python 3.8.12 (default, Sep  7 2021, 22:14:08)
[Clang 12.0.0 (clang-1200.0.32.29)]
```

### OS / Environment

Mac 10.15.7
